### ISSUE TITLE
 "cannot find loader for this FITS file" error

### ISSUE THREAD

**Issue by astrophysaxist** [imageio/imageio#636]
Hi, I was looking at the `imageio` module and I see that the FITS format is supposedly supported ([here](https://imageio.readthedocs.io/en/stable/formats.html)). But, I get an error when I try to use `imageio.get_reader()` on a file without the `format` keyword set to fits. So, I guess there's some mismatch in how `imagio` is trying to automatically determine the file type.

For a minimum reproducible example, you can stand up this environment:
```
conda create -n testio -c conda-forge scikit-image astropy imageio-ffmpeg pims python=3.7
```

After activating the environment, in Python generate a fake image and try reading it in:
```
import numpy as np
from astropy.io import fits
from astropy.modeling.models import Sersic2D
import imageio
xx, yy = np.meshgrid(np.arange(512),np.arange(512))
mod = Sersic2D(amplitude = 1, r_eff = 25, n=4, x_0=50,y_0=50, ellip=.5, theta=-1)
img = mod(xx, yy)
phdu = fits.PrimaryHDU()
ihdu = fits.ImageHDU(img)
hdul = fits.HDUList([phdu,ihdu])
hdul.writeto('test.fits')
im = imageio.get_reader('test.fits')
```
The last line fails for me with the following error:
```
Traceback (most recent call last):
  File "/local/data/home/[user]/.conda/envs/testio/lib/python3.7/site-packages/imageio/plugins/pillow.py", line 669, in pil_try_read
    im.getdata()[0]
  File "/local/data/home/[user]/.conda/envs/testio/lib/python3.7/site-packages/PIL/Image.py", line 1292, in getdata
    self.load()
  File "/local/data/home/[user]/.conda/envs/testio/lib/python3.7/site-packages/PIL/ImageFile.py", line 327, in load
    raise OSError(f"cannot find loader for this {self.format} file")
OSError: cannot find loader for this FITS file
```

But, I noticed if I give the format like the following line, the `get_reader()` call succeeds:
```
im = imageio.get_reader('test.fits',format='fits')
```

**Comment by FirefoxMetzger** [comment]
Interesting finding. I can reproduce this behavior on my end as well (bare python 3.8)

What is happening is that ImageIO chooses `pillow` as a matching backend, because it claims that it can handle FITS. This later fails when pillow is asked to actually read the file (as can be seen in the stack trace); maybe due to a missing feature in pillow's FITS reader? - not sure.

I think the solution here is to change priorities and make ImageIO prefer the `fits` backend (if present) over `pillow` for `.fits` files. This is an easy change, so I've marked this as a good first issue.

@astrophysaxist Would you be interested in contributing this change? I can review it and help you through the process.

**Comment by astrophysaxist** [comment]
@FirefoxMetzger thanks for the response. Sure I could help contribute to this change. I'd need a bit more context (where these changes have to occur, etc) to be of any use. But I'm open to that.

**Comment by FirefoxMetzger** [comment]
Awesome!

Plugins live in the `imageio/plugins` folder, and are imported in [this `__init__.py`](https://github.com/imageio/imageio/blob/master/imageio/plugins/__init__.py). The order in which the different plugins are imported reflects the order of priority over plugins (with the most important one coming first). In other words, if multiple backends can potentially read a format, the one that is associated with the plugin that was imported first will be called to do the loading.

The required change would be to move the `import fits` call up in this hierarchy so that it comes before `import pillow`. Additionally, it would be very nice, if you could turn your MWE into a unit-test (located under `tests/test_fits.py`) to avoid accidental regression later on.

For this, you will likely want to setup a developer environment for imageIO by forking this repo, cloning the fork, and installing imageio as an editable version via `pip install -e .[dev]` (inside the git repo). You can - of course - do this from within a conda environment as well.

Does this help to get you started?

**Comment by astrophysaxist** [comment]
Hi @FirefoxMetzger , yes! That's exactly what I needed to know. I'll fork and get started on a feature branch, and check in if I have any questions.

**Comment by astrophysaxist** [comment]
@FirefoxMetzger I got a branch working with the change to imageio/plugins/__init__.py and a new unit test. Which branch on imageio/imageio should I submit the pull request to? I wasn't sure if I should submit to the master branch or not.

**Comment by FirefoxMetzger** [comment]
@astrophysaxist Awesomesauce! Yes, master is the correct branch. 

In the description of your PR, could you please somewhere mention this issue via `Closes: #636`?
